src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,120 @@
+const { expensesService, usersService } = require('./../services');
+const { expenseSchema } = require('./../schemas');
+const joi = require('joi');
+
+const getAll = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  const expenses = expensesService.getAll({
+    userId: +userId,
+    categories,
+    from,
+    to,
+  });
+
+  res.json(expenses);
+};
+
+const getOne = (req, res) => {
+  const { id } = req.params;
+
+  const expense = expensesService.getById(+id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.json(expense);
+};
+
+const create = (req, res) => {
+  const { error, value } = expenseSchema.validate(req.body, {
+    stripUnknown: true,
+  });
+
+  if (error) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const isUserExist = !!usersService.getById(value.userId);
+
+  if (!isUserExist) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newExpense = expensesService.create(value);
```

Comment:
Handle potential errors from the `expensesService.create` method, such as database errors or duplicate entries, and respond with an appropriate status code and message.